compiler/hie-core/src/Development/IDE/Core/FileStore.hs

```diff
@@ -74,8 +74,8 @@ makeLSPVFSHandle lspFuncs = VFSHandle
    }
 
 
--- | Get the contents of a file, either dirty (if the buffer is modified) or from disk.
-type instance RuleResult GetFileContents = (FileVersion, StringBuffer)
+-- | Get the contents of a file, either dirty (if the buffer is modified) or Nothing to mean use from disk.
+type instance RuleResult GetFileContents = (FileVersion, Maybe StringBuffer)
```

Comment:
`FileVersion` already has a case for files that are read from disk where we use the modification time instead of the LSP version. Should we make a new type here so that we only have two cases, i.e., modification time + no content and LSP version + content, instead of four?

Reply:
We generally do very different things with the modtime vs the content, so I'm inclined to keep them very separate.